Products/zms/_xmllib.py

```diff
@@ -466,12 +466,16 @@ def toCdata(self, s, xhtml=False):
 
   # Return Text in CDATA.
   elif s is not None:
-    if isinstance(s, bytes):
-      s = s.decode('utf-8')
-    # Hack for invalid characters
-    s = s.replace(chr(30), '')
-    # Hack for nested CDATA
-    s = re.compile(r'\<\!\[CDATA\[(.*?)\]\]\>').sub(r'<!{CDATA{\1}}>', s)
+    try:

```

Comment:
Should not just try/except exceptions...

Reply:
But?